### PR TITLE
libgpg-error: update to 1.38; use new gen-lock-obj.sh.

### PR DESCRIPTION
**Pull request by rmccask**
gen-lock-obj.sh was supposed to be in 1.38 but was left of the of the tar file
so I added it back.

### CODE REVIEW

**Comment by pullmoll** [comment]
You can find that file online at e.g. https://github.com/gpg/libgpg-error.
The link to the file is https://github.com/gpg/libgpg-error/raw/master/src/gen-lock-obj.sh
If you add it to the distfiles and also add the checksum, you don't have to vendor it.

**Comment by sgn** [comment]
> You can find that file online at e.g. https://github.com/gpg/libgpg-error.
> The link to the file is https://github.com/gpg/libgpg-error/raw/master/src/gen-lock-obj.sh
> If you add it to the distfiles and also add the checksum, you don't have to vendor it.

:+1: +1000

> Or even better yet use the original source: http://git.gnupg.org/cgi-bin/gitweb.cgi?p=libgpg-error.git;a=blob_plain;f=src/gen-lock-obj.sh;hb=HEAD

Better link, fetch a fixed revision:

https://git.gnupg.org/cgi-bin/gitweb.cgi?p=libgpg-error.git;a=blob_plain;f=src/gen-lock-obj.sh;h=13858cfbf6c5f69f5f5fd64cd6fcc6a6c80eca3b

> 
> IIRC you need to `create_wrksrc=y` and add `build_wrksrc="${pkgname}-${version}"` because the file can then be found in `${wrksrc}/../gen-lock-obj.sh`. But please take it with a grain of salt and try :)

Does `skip_extraction` work better?

**Comment by rmccask** [comment]
I am now fetching the file as a 2nd entry in `distfiles` and copying it to the `wrksrc` directory.  I went with `skip_extraction` in this case.

File: srcpkgs/libgpg-error/template [hunk 1]
```diff
@@ -1,28 +1,25 @@
...
 revision=1
 build_style=gnu-configure
 short_desc="Library for error values used by GnuPG component"
 maintainer="Orphaned <orphan@voidlinux.org>"
 license="LGPL-2.1-or-later, GPL-2.0-or-later"
 homepage="https://www.gnupg.org"
-distfiles="https://www.gnupg.org/ftp/gcrypt/${pkgname}/${pkgname}-${version}.tar.bz2"
-checksum=b32d6ff72a73cf79797f7f2d039e95e9c6f92f0c1450215410840ab62aea9763
+distfiles="https://www.gnupg.org/ftp/gcrypt/${pkgname}/${pkgname}-${version}.tar.bz2
+	https://git.gnupg.org/cgi-bin/gitweb.cgi?p=libgpg-error.git;a=blob_plain;f=src/gen-lock-obj.sh;h=13858cfbf6c5f69f5f5fd64cd6fcc6a6c80eca3b>gen-lock-obj.sh"
```
Comment (pullmoll) [inline srcpkgs/libgpg-error/template:11]: Nitpicking: variable value indentation is generally one blank, not tab stop.
Same for the checksum below.

File: srcpkgs/libgpg-error/template [hunk 2]
```diff
@@ -1,28 +1,25 @@
...
-	local lock_obj_pub=src/syscfg/lock-obj-pub.${XBPS_CROSS_TRIPLET#*-}.h
-	if [ "$CROSS_BUILD" ]; then
-		$CC -DHAVE_CONFIG_H -I. -pthread -o gen-posix-lock-obj src/gen-posix-lock-obj.c
-		qemu-${XBPS_TARGET_QEMU_MACHINE}-static -L ${XBPS_CROSS_BASE} \
-			gen-posix-lock-obj > $lock_obj_pub
-		msg_normal "Generated $lock_obj_pub ...\n"
-		cat $lock_obj_pub
-	fi
+pre_configure() {
+	cp -p ${XBPS_SRCDISTDIR}/${pkgname}-${version}/gen-lock-obj.sh ${wrksrc}/src
```
Comment (pullmoll) [inline srcpkgs/libgpg-error/template:21]: LGTM

File: srcpkgs/libgpg-error/template (hunk 1 above)
Comment (rmccask) [reply srcpkgs/libgpg-error/template]: I have made the changes.

**Comment by pullmoll** [comment]
Ok so far but I do not see the `gen-lock-obj.sh` call in the log. Is it silent? Or is still something missing?

**Comment by rmccask** [comment]
The relevant line is: `configure: generated src/lock-obj-pub.native.h` and only if cross compiling.

**Pull request merged**